src/core/net/srp_server.cpp

```diff
@@ -1324,6 +1324,50 @@ otError Server::Service::SetFullName(const char *aFullName)
     return error;
 }
 
+otError Server::Service::GetNextTxtEntry(otDnsTxtIterator &aIterator, otDnsTxtEntry &aTxtEntry) const
+{
+    otError error = OT_ERROR_NONE;
+
+    for (uint16_t i = aIterator; i < mTxtLength;)
+    {
+        otDnsTxtEntry entry;
+        uint8_t       length = mTxtData[i++];
+
+        entry.mKey         = reinterpret_cast<const char *>(mTxtData + i);
+        entry.mKeyLength   = length;
+        entry.mValue       = nullptr;
+        entry.mValueLength = 0;
+
+        for (uint8_t j = 0; j < length; ++j)
+        {
+            if (mTxtData[i + j] == kKeyValueSeparator)
```

Comment:
```suggestion
            if (mTxtData[i + j] == kKeyValueSeparator)
```

Can we remove `mKeyLength` field but replace each `kKeyValueSeparator` to `\0` when the mTxtData is stored?

This would allow us to keep the `mKey` as a null-terminated string, which I think is better.

Reply:
That is the original idea in my mind, but there could be no `kKeyValueSeparator` for a TXT entry:

> If there is no '=' in a DNS-SD TXT record string, then it is a
   boolean attribute, simply identified as being present, with no value.

https://tools.ietf.org/html/rfc6763#section-6.4

Reply:
I am fine with us specifcing the keylength as well.

If we want to make `mKey` be null-terminted, one idea can be to use  a more compelx `iterator` type (e.g., a struct containing a `index` and a `char`). 
Basically when we replace a char in `mTxtData` with `\0` (to make it null terminted), we save the replaced char in the `Iterator` itself and then on next call we replace it back. I think this may work but I see that it can get complex though.

Reply:
I also thought about it but gave up because it seems too dirty :)